diff -ura a/Configurations/15-android.conf b/Configurations/15-android.conf
--- a/Configurations/15-android.conf	2025-09-30 14:37:40.000000000 +0200
+++ b/Configurations/15-android.conf	2025-11-02 01:00:52.365858682 +0100
@@ -35,7 +35,7 @@
             my $ndk_src_props = "$ndk/source.properties";
             my $is_ndk = -f $ndk_src_props;
             if ($is_ndk == $is_standalone_toolchain) {
-                die "\$ANDROID_NDK_ROOT=$ndk is invalid";
+                # die "\$ANDROID_NDK_ROOT=$ndk is invalid";
             }
             $ndk = canonpath($ndk);
 
diff -ura a/crypto/ec/asm/ecp_nistz256-armv4.pl b/crypto/ec/asm/ecp_nistz256-armv4.pl
--- a/crypto/ec/asm/ecp_nistz256-armv4.pl	2025-09-30 14:37:40.000000000 +0200
+++ b/crypto/ec/asm/ecp_nistz256-armv4.pl	2025-11-02 01:02:54.055857736 +0100
@@ -81,7 +81,7 @@
 die "insane number of elements" if ($#arr != 64*16*37-1);
 
 $code.=<<___;
-.rodata
+.section .rodata
 .globl	ecp_nistz256_precomputed
 .type	ecp_nistz256_precomputed,%object
 .align	12
diff -ura a/providers/implementations/rands/seeding/rand_unix.c b/providers/implementations/rands/seeding/rand_unix.c
--- a/providers/implementations/rands/seeding/rand_unix.c	2025-09-30 14:37:40.000000000 +0200
+++ b/providers/implementations/rands/seeding/rand_unix.c	2025-11-02 01:02:12.075858065 +0100
@@ -352,7 +352,7 @@
      * Note: Sometimes getentropy() can be provided but not implemented
      * internally. So we need to check errno for ENOSYS
      */
-#  if !defined(__DragonFly__) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#  if !defined(__DragonFly__) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__ANDROID__)
 #    if defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux)
     extern int getentropy(void *buffer, size_t length) __attribute__((weak));
 
