diff -ur openssl-3.5.4/Configurations/15-android.conf openssl/Configurations/15-android.conf
--- openssl-3.5.4/Configurations/15-android.conf	2025-09-30 14:37:40.000000000 +0200
+++ openssl/Configurations/15-android.conf	2025-10-19 14:39:31.662476518 +0200
@@ -8,7 +8,7 @@
 
     my $android_ndk = {};
     my %triplet = (
-        arm    => "arm-linux-androideabi",
+        arm    => "arm-eabi",
         arm64  => "aarch64-linux-android",
         mips   => "mipsel-linux-android",
         mips64 => "mips64el-linux-android",
@@ -35,7 +35,7 @@
             my $ndk_src_props = "$ndk/source.properties";
             my $is_ndk = -f $ndk_src_props;
             if ($is_ndk == $is_standalone_toolchain) {
-                die "\$ANDROID_NDK_ROOT=$ndk is invalid";
+                # die "\$ANDROID_NDK_ROOT=$ndk is invalid";
             }
             $ndk = canonpath($ndk);
 
@@ -161,7 +161,7 @@
             }
             $android_ndk = {
                 cflags   => $cflags . $sysroot,
-                cppflags => $cppflags,
+                cppflags => $cppflags . "-DOPENSSL_USE_IPV6=0",
                 bn_ops   => $arch =~ m/64$/ ? "SIXTY_FOUR_BIT_LONG"
                                             : "BN_LLONG",
             };
diff -ur openssl-3.5.4/crypto/bio/bss_dgram.c openssl/crypto/bio/bss_dgram.c
--- openssl-3.5.4/crypto/bio/bss_dgram.c	2025-09-30 14:37:40.000000000 +0200
+++ openssl/crypto/bio/bss_dgram.c	2025-10-19 14:40:55.442475877 +0200
@@ -102,6 +102,10 @@
 #  define BIO_CMSG_LEN(x) CMSG_LEN(x)
 # endif
 
+#if __ANDROID_API__ < 5
+#undef IPV6_PKTINFO
+#endif
+
 # if   M_METHOD == M_METHOD_RECVMMSG   \
     || M_METHOD == M_METHOD_RECVMSG    \
     || M_METHOD == M_METHOD_WSARECVMSG
diff -ur openssl-3.5.4/crypto/ec/asm/ecp_nistz256-armv4.pl openssl/crypto/ec/asm/ecp_nistz256-armv4.pl
--- openssl-3.5.4/crypto/ec/asm/ecp_nistz256-armv4.pl	2025-09-30 14:37:40.000000000 +0200
+++ openssl/crypto/ec/asm/ecp_nistz256-armv4.pl	2025-10-19 14:42:06.669141997 +0200
@@ -81,7 +81,7 @@
 die "insane number of elements" if ($#arr != 64*16*37-1);
 
 $code.=<<___;
-.rodata
+.section .rodata
 .globl	ecp_nistz256_precomputed
 .type	ecp_nistz256_precomputed,%object
 .align	12
diff -ur openssl-3.5.4/providers/implementations/rands/seeding/rand_unix.c openssl/providers/implementations/rands/seeding/rand_unix.c
--- openssl-3.5.4/providers/implementations/rands/seeding/rand_unix.c	2025-09-30 14:37:40.000000000 +0200
+++ openssl/providers/implementations/rands/seeding/rand_unix.c	2025-10-19 14:45:27.325807125 +0200
@@ -352,7 +352,7 @@
      * Note: Sometimes getentropy() can be provided but not implemented
      * internally. So we need to check errno for ENOSYS
      */
-#  if !defined(__DragonFly__) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#  if !defined(__DragonFly__) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__ANDROID__)
 #    if defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux)
     extern int getentropy(void *buffer, size_t length) __attribute__((weak));
 
